### ISSUE TITLE
Query Regarding EER Calculation

### ISSUE THREAD

**Issue by minyoung-kyoung** [Jiang-Yidi/TS-TalkNet#1]
Hello, thank you for sharing your valuable code :)
How did you calculate the EER, and could you share the relevant source code? (I understand EER is for speaker recognition...)

**Comment by Jiang-Yidi** [comment]
Hi, thanks for your question.

Sorry that I only upload the code on AVA dataset.
The relevant code to calculate AP,AUC, EER for ASW dataset is:

`from sklearn import metrics`
`from scipy.optimize import brentq`
`from scipy.interpolate import interp1d`
`ACC = numpy.round(metrics.accuracy_score(all_labels, predlabels) * 100, 2)`
`AP = numpy.round(metrics.average_precision_score(all_labels, predScores) * 100, 2)`
`fpr, tpr, thresholds = metrics.roc_curve(all_labels, predScores, pos_label=1)`
`AUC = numpy.round(metrics.auc(fpr, tpr) * 100, 2)`
`EER = numpy.round(brentq(lambda x : 1. - x - interp1d(fpr, tpr)(x), 0., 1.) * 100, 2)`

I follow the metric of baseline paper [Look Who’s Talking: Active Speaker Detection in the Wild](https://arxiv.org/pdf/2108.07640.pdf), which released the ASW dataset.

Hope these can help.

**Comment by DevKiHyun** [comment]
Hi @Jiang-Yidi ,

In your code, predlabels and predScores is Same?
If it is not same, then predlabels is predict output from model and its shape is [batch_size, 2(==n_class)]

and predScores is predScores = predlabels[:,1] and its shape is [batch_size, 1] right?

**Comment by Jiang-Yidi** [comment]
Hi @DevKiHyun ,

They are not the same.
you can refer the code in "loss.py"

`predScore = F.softmax(x, dim = -1)`
`predLabel = torch.round(F.softmax(x, dim = -1))[:,1]`

**Comment by DevKiHyun** [comment]
Hi Yidi!
Thanks!
